turtlebot_follower/launch/follower.launch

```diff
@@ -2,29 +2,26 @@
   The turtlebot people (or whatever) follower nodelet.
  -->
 <launch>
-  <arg name="simulation" default="false"/>
+  <arg name="simulation"    default="false"/>
+  <arg name="generate_goal" default="true"/> <!-- Use the default goal point generation -->
+  
+  <!-- Generate Goal: Load camera and centroid goal generation nodelets -->
+  <!-- Else: Assume goal generation is loaded elsewhere -->
+  <include if="$(arg generate_goal)" file="$(find turtlebot_follower)/launch/includes/centroid.launch">
+	<arg name="simulation"  value="$(arg simulation)"/>
+  </include>
+  
+  <node pkg="nodelet" type="nodelet" name="follower_nodelet_manager" args="manager"/>
+  
   <group unless="$(arg simulation)"> <!-- Real robot -->
     <include file="$(find turtlebot_follower)/launch/includes/velocity_smoother.launch.xml">
       <arg name="nodelet_manager"  value="/mobile_base_nodelet_manager"/>
       <arg name="navigation_topic" value="/cmd_vel_mux/input/navi"/>
     </include>
-
-    <include file="$(find turtlebot_bringup)/launch/3dsensor.launch">
-      <arg name="rgb_processing"                  value="true"/>  <!-- only required if we use android client -->
-      <arg name="depth_processing"                value="true"/>
-      <arg name="depth_registered_processing"     value="false"/>
-      <arg name="depth_registration"              value="false"/>
-      <arg name="disparity_processing"            value="false"/>
-      <arg name="disparity_registered_processing" value="false"/>
-      <arg name="scan_processing"                 value="false"/>
-    </include>
   </group>
   <group if="$(arg simulation)">
-    <!-- Load nodelet manager for compatibility -->
-    <node pkg="nodelet" type="nodelet" ns="camera" name="camera_nodelet_manager" args="manager"/>
-
     <include file="$(find turtlebot_follower)/launch/includes/velocity_smoother.launch.xml">
-      <arg name="nodelet_manager"  value="camera/camera_nodelet_manager"/>
+      <arg name="nodelet_manager"  value="/follower_nodelet_manager"/>
```

Comment:
Hi @tfoote -- good catch on the leading slash on the manager name.

The reason I created an independent manager to run the follower nodelet itself, rather than running it in the camera_nodelet_manager like it was previously, is because when you abstract away the goal generation, you can no longer guarantee that the camera_nodelet_manager will still exist.  

The centroid-based goal generation implemented here launches the camera in the standard way, and runs in the same nodelet manager as that camera to efficiently avoid copying the camera data stream.  However, the RealSense Person goal generation nodelet linked above relies on the RealSense Person package, which is not turtlebot-specific, cannot launch the camera in the standard way (it needs to configure additional dimension and framerate options that are not passed through the standard turtlebot_bringup camera launcher), and does not call its nodelet manager camera_nodelet_manager.

Further, since _any_ nodelet could be used to generate the goals, a totally different sensor mechanism that does not rely on cameras at all (e.g., lidar) could be used to tell the robot where to follow.